tests/test_unit/test_uncertainty_evaluators.py

```diff
@@ -1,159 +1,126 @@
 """Chemprop unit tests for chemprop/uncertainty/uncertainty_evaluator.py"""
-from unittest import TestCase
+import uuid
+
 import numpy as np
+import pytest
+
+from chemprop.uncertainty.uncertainty_evaluator import build_uncertainty_evaluator
+
+
+@pytest.fixture(params=["regression", "classification", "multiclass"])
+def dataset_type(request):
+    return request.param
+
+
+@pytest.fixture(params=["nll", "miscalibration_area", "ence", "spearman"])
+def regression_metric(request):
+    return request.param
+
+
+@pytest.fixture(params=["auc", "prc-auc", "accuracy", "f1", "mcc"])
+def classification_metric(request):
+    return request.param
+
+
+@pytest.fixture(params=["cross_entropy", "accuracy", "f1", "mcc"])
+def multiclass_metric(request):
+    return request.param
+
+
+@pytest.fixture
+def nll_regression_evaluator():
+    return build_uncertainty_evaluator("nll", None, "ensemble", "regression", "mse", None)
+
+
+@pytest.fixture
+def nll_classification_evaluator():
+    return build_uncertainty_evaluator(
+        "nll", None, "classification", "classification", "binary_cross_entropy", None
+    )
+
+
+@pytest.fixture
+def miscal_regression_evaluator():
+    return build_uncertainty_evaluator(
+        "miscalibration_area", None, "ensemble", "regression", "mse", None
+    )
+
+
+@pytest.fixture
+def spearman_evaluator():
+    return build_uncertainty_evaluator("spearman", None, "ensemble", "regression", "mse", None)
+
+
+def test_build_regression_metric(regression_metric):
+    assert build_uncertainty_evaluator(regression_metric, None, None, "regression", None, None)
```

Comment:
Previously, this test checked to see if the output was an `UncertaintyEvaluator`. Now it's changed to be checking just that it's not None or False right?

I can't think of a corner case where the distinction would matter I suppose. So I think it's okay.

Reply:
Also this request structure is interesting. Wasn't familiar with it previously.